ratbag_emu/device.py

```diff
@@ -227,15 +227,10 @@ def simulate_action(self, action: Dict[str, Any], type: int = None) -> None:
         elif action['type'] == ActionType.BUTTON:
             self._simulate_action_xy(action, packets, report_count)
 
-        def send_packets() -> None:
-            nonlocal packets
-            s = sched.scheduler(time.time, time.sleep)
-            next_time = 0.0
-            for packet in packets:
-                s.enter(next_time, 1, self.send_hid_action,
-                        kwargs={'action': packet})
-                next_time += 1 / self.report_rate
-            s.run()
-
-        sim_thread = threading.Thread(target=send_packets)
-        sim_thread.start()
+        s = sched.scheduler(time.time, time.sleep)
+        next_time = 0.0
+        for packet in packets:
+            s.enter(next_time, 1, self.send_hid_action,
+                    kwargs={'action': packet})
+            next_time += 1 / self.report_rate
+        s.run()
```

Comment:
I'd like to have this change in a separate commit: it took me a while to figure out that you actually made the `simulate_action` synchronous now.

Reply:
Will do.